Integrate Weblate into monerujo + remove fuzzy translations

lucasmz-dev opened the pull request:
https://hosted.weblate.org/projects/monerujo

Being dealt with in a separate branch that we can test everything as safely as possible.
After this is merged, it will be changed to use an automatic GitHub PR + automatic webhooks so that pull is automatic.

Let's see how Weblate behaves with monerujo strings.

Contains #960 
Closes #958

lucasmz-dev committed:
Remove tools:ignore flags

lucasmz-dev committed:
Disable linting for MissingTranslation

lucasmz-dev committed:
Remove translation of strings that were changed

Removes strings from translations, where the string was translated and the source string was later changed, but the translation never invalidated

May potentially not include all strings, the process to find them was to look into Git Blame and see weird dates and what was changed.

lucasmz-dev committed:
Remove english duplicates from translations

We don't need this, we ignore the MissingTranslation flag.
It also means it fallbacks to another variant if available, instead of always being english.

lucasmz-dev committed:
Update credits in translation files

Quick find and replace

lucasmz-dev committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (368 of 368 strings)

Translation: monerujo/App - Main
Translate-URL: https://hosted.weblate.org/projects/monerujo/app-main/pt_BR/

lucasmz-dev committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (20 of 20 strings)

Translation: monerujo/App - Help
Translate-URL: https://hosted.weblate.org/projects/monerujo/app-help/pt_BR/

lucasmz-dev commented:
@m2049r please do not modify source strings from now on! it's fine if you do but don't change translations! It's better to remove them, maintenance wise. It's IMO easier for us to translate things again than to have to go through every string just to see that something might be wrong, and it becomes confusing because sometimes things can be well... adaptations. If you can get those basic changes that disable linting for MissingTranslation, that would be nice.

I still need to figure out how to build to see how things are looking, but it seems OK ATM. These CDATA strings are kind of bad, they show up in a very raw format in Weblate and so the translator still needs to deal with things like \s, weird things like that...

lucasmz-dev committed:
Point README translation to Weblate

weblate committed:
Update translation files

Updated by "Cleanup translation files" add-on in Weblate.

Translation: monerujo/App - Main
Translate-URL: https://hosted.weblate.org/projects/monerujo/app-main/

m2049r commented on app/src/main/res/values-cat/strings.xml:
```diff
@@ -30,53 +30,31 @@
     <string name="info_paymentid_integrated">&#x2714; ID de pagment integrat</string>
     <string name="info_prepare_tx">Preparant la seva transacció</string>
 
-    <string name="label_send_progress_xmrto_create">Creant ordre</string>
```
why are some used and translated strings removed?

lucasmz-dev replied:
I'll have to take a closer look, but what I did was remove any translated strings that have not been modified after the source string was, for safety, basically. This could have gotten rid of a few valid strings.

m2049r replied:
we should leave all translated strings in, removing only english ones

lucasmz-dev commented:
Um... weblate deleted the monerujo project 😀
i am very confused
sending them an email

lucasmz-dev commented:
@m2049r they restored it ^^
might've been my fault I might have accidentally clicked to ask for approval before the time 😬
you might need to accept an invitation

m2049r committed:
Translated using Weblate (Spanish)

Currently translated at 80.0% (4 of 5 strings)

Translation: monerujo/App - About
Translate-URL: https://hosted.weblate.org/projects/monerujo/app-about/es/

Commit:
Translated using Weblate (Spanish)

Currently translated at 80.0% (4 of 5 strings)

Translation: monerujo/App - About
Translate-URL: https://hosted.weblate.org/projects/monerujo/app-about/es/

lucasmz-dev committed:
Merge branch 'master' of https://github.com/m2049r/xmrwallet into weblate

lucasmz-dev committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 99.7% (371 of 372 strings)

Translation: monerujo/App - Main
Translate-URL: https://hosted.weblate.org/projects/monerujo/app-main/pt_BR/

lucasmz-dev commented:
I went over all strings I didn't do, and flagged anything that seemed weird with a comment, will be looking into it in Git history soon to see if it was just an adaptation or if it was a changed string or something
e.g. Node -> Network

lucasmz-dev committed:
Add language view + typo fix to README

also changes the link with an engage link instead of a regular project link

lucasmz-dev committed:
Make "Scan" translatable

lucasmz-dev committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 99.7% (371 of 372 strings)

Translation: monerujo/App - Main
Translate-URL: https://hosted.weblate.org/projects/monerujo/app-main/pt_BR/

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (5 of 5 strings)

Translation: monerujo/App - About
Translate-URL: https://hosted.weblate.org/projects/monerujo/app-about/es/

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (20 of 20 strings)

Translation: monerujo/App - Help
Translate-URL: https://hosted.weblate.org/projects/monerujo/app-help/es/

Commit:
Translated using Weblate (Spanish)

Currently translated at 99.7% (371 of 372 strings)

Translation: monerujo/App - Main
Translate-URL: https://hosted.weblate.org/projects/monerujo/app-main/es/

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (372 of 372 strings)

Translation: monerujo/App - Main
Translate-URL: https://hosted.weblate.org/projects/monerujo/app-main/es/

lucasmz-dev committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (372 of 372 strings)

Translation: monerujo/App - Main
Translate-URL: https://hosted.weblate.org/projects/monerujo/app-main/pt_BR/

lucasmz-dev commented:
Hey, @m2049r
Could you build a debug build for this branch? It'd help a lot!

lucasmz-dev committed:
Remove some extra leftover english duplicates

Detected by Weblate

lucasmz-dev committed:
Handle "Changed to" strings

lucasmz-dev committed:
Fix random broken translation

lucasmz-dev committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (372 of 372 strings)

Translation: monerujo/App - Main
Translate-URL: https://hosted.weblate.org/projects/monerujo/app-main/pt_BR/

lucasmz-dev committed:
Fix translation inconscienties found manually in PT-BR that are present globally

AKA strings that were changed on source but not removed from translations, all that

lucasmz-dev commented:
I think this might be ready to merge! Need to check if everything seems alright, but after that, and being merged, we can point things properly to this repository and after that ask for libre hosting.

<s> EDIT: actually I need to merge the other translations... let's see</s>

lucasmz-dev committed:
Turkish + French translations manual import

Squashed commit of the following:

commit 65fd46539db726a2557b6ff3af56c12df046d5ac
Author: cardpuncher <64688741+cardpuncher@users.noreply.github.com>
Date:   Sun Oct 6 10:51:31 2024 +0300

    Update help.xml

commit 6d5190f776fd45f51ccb4428a4dcc16b4620e988
Author: cardpuncher <64688741+cardpuncher@users.noreply.github.com>
Date:   Sun Oct 6 10:38:38 2024 +0300

    All errors should have been cleared now

    Crossing fingers, click & pray.

commit 0ae3647f113d4f8ed1c51c19134a8daa0453e5f7
Author: cardpuncher <64688741+cardpuncher@users.noreply.github.com>
Date:   Sun Oct 6 10:32:00 2024 +0300

    Still on it

commit f8346a9b2a326eab88fb33e030fbccf9933139ff
Author: cardpuncher <64688741+cardpuncher@users.noreply.github.com>
Date:   Sun Oct 6 10:29:52 2024 +0300

    Error hunting goes on

    Hopefully there aren't many left.

commit 08bf318e843accb54d34e06187c919bb53b9f14d
Author: cardpuncher <64688741+cardpuncher@users.noreply.github.com>
Date:   Sun Oct 6 10:11:02 2024 +0300

    Squash some syntax errors, there are others

commit df9ea45220320ec800e54d2129c1a3d63b4e2d73
Author: cardpuncher <64688741+cardpuncher@users.noreply.github.com>
Date:   Sun Oct 6 09:41:33 2024 +0300

    Update French translations

    and fix a few typos in the process.

commit 3ea605fac2122d9d2be2b55d18a5f3aa0118e8c5
Author: cardpuncher <64688741+cardpuncher@users.noreply.github.com>
Date:   Sat Oct 5 23:16:56 2024 +0300

    Complete Turkish translation

commit f04e3dd4da6b86495e72ff03f8172e13b50d7a0f
Merge: abae1f3 3e56d5a
Author: cardpuncher <64688741+cardpuncher@users.noreply.github.com>
Date:   Sat Oct 5 23:15:38 2024 +0300

    Merge branch 'm2049r:master' into master

commit abae1f3ac644349d951f47a1d867185b9fe87efe
Author: cardpuncher <64688741+cardpuncher@users.noreply.github.com>
Date:   Fri Sep 20 12:36:27 2024 +0300

    Yet another copy & paste error... :/

commit b1118aea01db1ba3b60bffc73de38f84442723c9
Author: cardpuncher <64688741+cardpuncher@users.noreply.github.com>
Date:   Fri Sep 20 12:29:17 2024 +0300

    Fix a dumb copy & paste error

    Hopefully this will silence the error.

commit 0ae7c8c514d69cb691c20ef8a95f911637ae3249
Author: cardpuncher <64688741+cardpuncher@users.noreply.github.com>
Date:   Fri Sep 20 12:18:10 2024 +0300

    Update French & Turkish translations

    Mainly for the Sidekick part, also added a few missing strings to the French translation and reverted to use the original "Ledger" word in the Turkish one.

Co-authored-by: cardpuncher <64688741+cardpuncher@users.noreply.github.com>

lucasmz-dev committed:
Update swedish translation

Co-authored-by: Ordtrogen <ordtrogen@users.noreply.github.com>

m2049r commented on app/src/main/res/values-cat/help.xml:
```diff
@@ -109,50 +109,6 @@
         Aquest és el bloc de la cadena de Monero on la vostra transacció ha estat inclosa.
     ]]></string>
 
-    <string name="help_send"><![CDATA[
```
i still don't think we should be losing good translations

lucasmz-dev replied:
I mean I can just go over and just cherry-pick the barebones, but the reason why I went over with this is that we can't for sure know for every language that they are in fact "good translations"
Changing wording can mean very different things in different languages and affect what gender's used, word position, stuff like that, that's the reason I went over like this, the people who want to make these translations work out will and with Weblate any changes in the future can be easily fixed by translators and not go unnoticed

If you're fine with that, I'll do it

lucasmz-dev replied:
Weblate has a pretty nice graceful way to deal with "fuzzy" translations, they're marked as needing for editing and show the differences done in the source string, but it doesn't do anything about old commits before integration with it, that's basically what I tried to achieve here, in the end it'd be less work in the end to you and the translators IMO
A lot of translators aren't gonna have the energy/time/want to go through everything, and just wanna be able to translate right off the bat, I am like that for certain apps, especially when they have a messy situation anyway

RN my suggestion is: check #975 and I'll rebase this and then you can see if you want this, so that we can get weblate hosting at least since we'll need those to get things working at a basic level

m2049r commented:
if you want, i can look through all translations and remove the non-translated stuff. so this PR would be the barebones of weblate enablement?
not sure what's best

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (372 of 372 strings)

Translation: Monerujo/App - Main
Translate-URL: https://hosted.weblate.org/projects/monerujo/app-main/es/

lucasmz-dev committed:
Set untranslatable strings as so

Already set these as read-only in Weblate but better to do this in the source code

lucasmz-dev committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (372 of 372 strings)

Translation: monerujo/App - Main
Translate-URL: https://hosted.weblate.org/projects/monerujo/app-main/pt_BR/

lucasmz-dev committed:
Merge branch 'master' into weblate
